client/src/ActionsAndReducers/dbWargames/wargames_Reducer.ts

```diff
@@ -22,107 +32,121 @@ const initialState = {
   const tab = newState.currentTab
 
-  let selected: string
-  let curChannel: string
-  let index
-  let newForce
-  let newChannel
-  let channelIndex
-  let newParticipant
-
-  switch (action.type) {
-    case ActionConstant.ALL_WARGAME_NAMES_SAVED:
-      const originalList = action.payload || []
-      const anyList = originalList as any
-      // NOTE: we don't know why, but some SQLite files 
-      // can get corrupted, and are empty. Filter them
-      // out, since they're of no values
-      const safeList = anyList.filter((game: any) => {
-        if (game.wargameTitle || game.title) {
-          return true
-        } else {
-          console.warn('Warning Data not found for', game.name, ' potentially corrupt')
-          return false
-        }
-      })
-      newState.wargameList = safeList
-      return newState
-
-    case ActionConstant.SET_CURRENT_WARGAME:
-
-      newState.currentWargame = action.payload.name
-      newState.wargameTitle = action.payload.wargameTitle
-      newState.data = action.payload.data
-      newState.wargameInitiated = action.payload.wargameInitiated || false
-
-      return newState
-
-    case ActionConstant.SET_EXPORT_WARGAME:
-
-      newState.data = action.payload.data
-      newState.currentWargame = action.payload.name
-      newState.wargameTitle = action.payload.wargameTitle
-      newState.exportMessagelist = action.payload.exportMessagelist
-      newState.wargameInitiated = action.payload.wargameInitiated || false
-
-      return newState
-
-    case ActionConstant.SET_CURRENT_GAME_SETUP_TAB:
-      newState.currentTab = action.payload
-      break
-
-    case ActionConstant.SET_GAME_SETUP_DATA:
-      newState.data[tab] = { ...newState.data[tab], ...action.payload }
-      break
-
-    case ActionConstant.SET_TAB_SAVED:
-      newState.data[tab].dirty = false
-      break
-
-    case ActionConstant.SET_TAB_UNSAVED:
-      newState.data[tab].dirty = true
-      break
-
-    case ActionConstant.ADD_NEW_FORCE:
-
-      newForce = forceTemplate
-      newForce.name = action.payload.name
-      newForce.uniqid = action.payload.uniqid
-
-      newState.data[tab].forces.unshift(newForce)
-      break
-
-    case ActionConstant.SET_FORCE_COLOR:
-      newState.data[tab].forces.find((force: ForceData) => force.name === newState.data[tab].selectedForce.name).color = action.payload
-      break
-
-    case ActionConstant.SET_SELECTED_FORCE:
-      newState.data.forces.selectedForce = action.payload
-      break
-
-    case ActionConstant.ADD_NEW_CHANNEL:
+  const handler = handlers[action.type]
 
-      newChannel = channelTemplate
-      newChannel.name = action.payload.name
-      newChannel.uniqid = action.payload.uniqid
-      newState.data[tab].channels.push(newChannel)
-      break
-
-    case ActionConstant.SET_SELECTED_CHANNEL:
-      newState.data[tab].selectedChannel = action.payload
-      break
-
-    case ActionConstant.DELETE_SELECTED_CHANNEL:
```

Comment:
@lilitkarapetyan - if the `handler` is empty in the next next line, should we throw a warning/error?

Reply:
I will do add warrning.